packages/core/src/components/tooltip/tooltip.tsx

```diff
@@ -0,0 +1,201 @@
+import { createPopper } from '@popperjs/core'
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Listen,
+  Prop,
+  State,
+  Watch,
+  h,
+} from '@stencil/core'
+
+@Component({
+  tag: 'atom-tooltip',
+  styleUrl: 'tooltip.scss',
+})
+export class AtomTooltip {
+  private _popperInstance: ReturnType<typeof createPopper> = null
+  private _eventsToShow = ['mouseenter', 'focus']
+  private _eventsToHide = ['mouseleave', 'blur']
+  private _eventsToShowMobile = ['focus', 'click']
+  private _eventsToHideMobile = ['blur']
+  private _elementSelector: Element = null
+
+  @Element() el: HTMLElement
+
+  @State() open = false
+
+  /**
+   * Determines which element by a ID should listen to open tooltip.
+   */
+  @Prop() element: string
+
+  /**
+   * Determines the placement for attach tooltip.
+   */
+  @Prop() placement:
+    | 'auto'
+    | 'auto-start'
+    | 'auto-end'
+    | 'top'
+    | 'top-start'
+    | 'top-end'
+    | 'bottom'
+    | 'bottom-start'
+    | 'bottom-end'
+    | 'right'
+    | 'left' = 'top'
+
+  @Watch('placement')
+  placementShouldUpdatePopperInstance(newPlacement: typeof this.placement) {
+    this._popperInstance.setOptions((options) => ({
+      ...options,
+      placement: newPlacement,
+    }))
+
+    this._popperInstance.update()
+  }
+
+  /**
+   * Event is dispatched when:
+   * 1. Mouse leave event in selector
+   * 2. Click "Close" button when is mobile
+   */
+  @Event() atomClose: EventEmitter
+
+  /**
+   * Event is dispatched when:
+   * 1. Mouse enter event in selector
+   * 2. Click in element when is mobile
+   */
+  @Event() atomOpen: EventEmitter
+
+  @Listen('resize', { target: 'window' })
+  onResize() {
+    this.attachEvents()
+  }
+
+  connectedCallback() {
+    const selector = document.getElementById(this.element)
+
+    this._elementSelector = selector
+
+    this.attachEvents()
+
+    this._popperInstance = createPopper(selector, this.el, {
+      placement: this.placement,
+      modifiers: [
+        {
+          name: 'offset',
+          options: {
+            offset: [0, 4],
+          },
+        },
+        {
+          name: 'arrow',
+          options: {
+            padding: 16,
+          },
+        },
+      ],
+    })
+  }
+
+  disconnectedCallback() {
+    this.untachEvents()
+
+    this._popperInstance.destroy()
+  }
+
+  private isMobile() {
+    return window.matchMedia('(max-width: 768px)').matches
+  }
```

Comment:
(question) do you test it inside vue project?

Reply:
Yeah! it worked perfectly, better than react.
I ended up running into an issue that I'm investigating with SSR.

Reply:
https://github.com/juntossomosmais/atomium/assets/54173994/2fd68551-e663-431c-b5f9-2bf06aa1520d

Reply:
@MayaraRMA Is a good idea your validation here too

Reply:
@MayaraRMA @felipefialho you able to validate?